sources/playground/playground.sh

```diff
@@ -0,0 +1,595 @@
+in_command_block { print; next; }
+
+/^CQL Playground/  { print "\033[1;4m" $0 "\033[0m"; next }
+
+/^        / { print; next }
+/^    /     { print "\033[1m" $0 "\033[0m"; next }
+/^./        { print "\033[4m" $0 "\033[0m"; next }
+            { print }
+
+END { printf "\033[0;0m" }'
+}
+
+resolve_example_name_from_source() {
+    local source=$1
+
+    echo $source | sed 's|.*/\(.*\)\.sql$|\1|'
+}
+
+execute() {
+    local description="$1"
+    local command="$2"
+
+    shift
+
+    # if very verbose and description is not empty print the command:
+    if [[ $VERBOSITY_LEVEL -ge 3 && -n "$description" ]]; then
+        echo -e "\n# $description" | theme >&2
+    fi
+
+    if [[ $VERBOSITY_LEVEL -ge 2 && -n "$description" ]]; then
+        echo "COMMAND: $command" | theme >&2
+    fi
+
+    eval "$command"
+
+    if [[ $? -ne 0 ]]; then
+        echo -e "ERROR: The command failed" | theme >&2
+        echo -e "NOTE: If there is \"No such file or directory\", try with --rebuild" | theme
+
+        if [[ $VERBOSITY_LEVEL -lt 2 ]]; then
+            echo "COMMAND: $command" | theme >&2
+        fi
+
+        exit 1
+    fi
+}
+
+is_dependency_satisfied() {
+    for dependency in "$@"; do
+        case $dependency in
+            java)
+                type java >/dev/null 2>&1 \
+                && test -n "${JAVA_HOME}" \
+                && return 0 || return 1
+                ;;
+            lsqlite)
+                type luarocks >/dev/null 2>&1 \
+                && luarocks show lsqlite3 --porcelain >/dev/null 2>&1 \
+                && return 0 || return 1
+                ;;
+
+            cql_compiler) type "$CQL"  >/dev/null 2>&1 && return 0 || return 1 ;;
+            jq)           type jq      >/dev/null 2>&1 && return 0 || return 1 ;;
+            lua)          type lua     >/dev/null 2>&1 && return 0 || return 1 ;;
+            dot)          type dot     >/dev/null 2>&1 && return 0 || return 1 ;;
+ 
+            *)
+                echo "Unknown dependency: $1";
+                exit 1
+                ;;
+        esac
+    done
+}
+
+guard_against_unsatisfied_required_dependencies() {
+    if ! is_dependency_satisfied cql_compiler; then
+        hello
+        exit 1
+    fi
+}
+
+initialize_example() {
+    local source=$1
+    local example_name=$2
+    local example_output_dir_relative="$SCRIPT_OUT_DIR_RELATIVE/$example_name"
+
+    execute "Create the output directory if it doesn't exist" \
+        "mkdir -p $example_output_dir_relative"
+    
+    execute "Clean the content of the output directory" \
+        "rm -rf $example_output_dir_relative/*"
+    
+    execute "Copy the orginal sql file" \
+        "cp $source $example_output_dir_relative/$example_name.sql.original"
+
+    execute "The 'preprocessed' output — Preprocess the sql file (e.g.: apply macros)" \
+        "cc --preprocess --language=c $example_output_dir_relative/$example_name.sql.original > $example_output_dir_relative/$example_name.sql.pre"
```

Comment:
Another interesting thing to do here is to enable cql error tracing optionally.  Look at `upgrade/errortrace.inc` -- adding it conditionally could be very useful.  It causes a printout of each fail point as the exception ripples to the top.

Reply:
Btw you don't have to do all of these additional things in this PR.